src/client/auth/mod.rs

```diff
@@ -262,12 +300,23 @@ impl Credential {
         };
 
         // Authenticate according to the chosen mechanism.
-        mechanism.authenticate_stream(conn, self).await
+        mechanism.authenticate_stream(conn, self, http_client).await
     }
 }
 
-pub(crate) fn generate_nonce() -> String {
+pub(crate) fn generate_nonce_bytes() -> Vec<u8> {
     let mut rng = rand::thread_rng();
-    let result: Vec<u8> = (0..32).map(|_| rng.gen()).collect();
+    (0..32).map(|_| rng.gen()).collect()
+}
+
+pub(crate) fn generate_nonce() -> String {
+    let result = generate_nonce_bytes();
     base64::encode(result.as_slice())
 }
+
+fn mac<M: Mac>(key: &[u8], input: &[u8]) -> Result<impl AsRef<[u8]>> {
+    let mut mac =
+        M::new_varkey(key).or_else(|_| Err(Error::unknown_authentication_error("SCRAM")))?;
```

Comment:
so this always will emit an error for SCRAM, but it may be used with other auth mechanisms.

Reply:
Oops, should have paid more attention to my copypasta

Reply:
FYI you haven't pushed any commits yet to address this (just in case you forgot, no rush if you haven't gotten to it).

Reply:
Sorry about that; just pushed now